CRM/Admin/Form/SettingTrait.php

```diff
@@ -227,7 +228,15 @@ protected function addFieldsDefinedInSettingsMetadata() {
           $this->addRadio($setting, $props['title'], [1 => ts('Yes'), 0 => ts('No')], CRM_Utils_Array::value('html_attributes', $props), '&nbsp;&nbsp;');
         }
         elseif ($add === 'add') {
-          $this->add($props['html_type'], $setting, $props['title'], $options);
+          if ($props['html_type'] === 'datepicker') {
+            if (isset($props['html_extra']['minDate'])) {
+              $props['html_extra']['minDate'] = date('Y-m-d', strtotime($props['html_extra']['minDate']));
+            }
+            if (isset($props['html_extra']['maxDate'])) {
+              $props['html_extra']['maxDate'] = date('Y-m-d', strtotime($props['html_extra']['maxDate']));
+            }
+          }
```

Comment:
What about moving the above two conditionals into the `CRM_Core_Form::add()` function?

Reply:
I'll take a look

Reply:
Now done